ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -8088,6 +8085,25 @@ static bool ggml_vk_instance_portability_enumeration_ext_available(const std::ve
     UNUSED(instance_extensions);
 }
 
+static bool ggml_vk_khr_cooperative_matrix_support(const vk::PhysicalDeviceProperties& props, const vk::PhysicalDeviceDriverProperties& driver_props) {
+    switch (props.vendorID) {
+    case VK_VENDOR_ID_INTEL:
+        // Intel drivers don't support coopmat properly yet
+        return false;
+    case VK_VENDOR_ID_AMD:
+        if (driver_props.driverID == vk::DriverId::eAmdProprietary || driver_props.driverID == vk::DriverId::eAmdOpenSource) {
+            // Workaround for AMD proprietary driver reporting support on all GPUs
+            const std::string name = props.deviceName;
+            return name.rfind("AMD Radeon RX 7", 0) == 0 ||    // RDNA 3 consumer GPUs
```

Comment:
Is there a way to write this so newer GPUs will be enabled by default?

Reply:
That would be good, but I couldn't think of an easy way. I'm hoping that AMD resolves this in their drivers, then we can just remove the check.

I'll try to figure out the proper way to report this to their Windows and amdvlk driver team.